src/Common/src/System/Net/ContextAwareResult.cs

```diff
@@ -140,9 +140,13 @@ internal ExecutionContext ContextCopy
         {
             get
             {
-                GlobalLog.Assert(!InternalPeekCompleted || (_flags & StateFlags.ThreadSafeContextCopy) != 0, "ContextAwareResult#{0}::ContextCopy|Called on completed result.", LoggingHash.HashString(this));
                 if (InternalPeekCompleted)
                 {
+                    if ((_flags & StateFlags.ThreadSafeContextCopy) == 0 && GlobalLog.IsEnabled)
```

Comment:
/cc @josguil 
This actually won't work: the original Desktop code will Debug.Assert() in these cases. I've opened #5265 to track this work.